client/api/src/notifications.rs

```diff
@@ -21,30 +21,49 @@
 use std::{
 	collections::{HashMap, HashSet},
 	pin::Pin,
-	sync::{Arc, Weak},
+	sync::Arc,
 	task::Poll,
 };
 
-use fnv::{FnvHashMap, FnvHashSet};
 use futures::Stream;
-use parking_lot::Mutex;
-use prometheus_endpoint::{register, CounterVec, Opts, Registry, U64};
-use sc_utils::mpsc::{tracing_unbounded, TracingUnboundedReceiver, TracingUnboundedSender};
-use sp_core::{
-	hexdisplay::HexDisplay,
-	storage::{StorageData, StorageKey},
-};
+
+use prometheus_endpoint::Registry as PrometheusRegistry;
+
+use sc_utils::pubsub::{channels::TracingUnbounded, Hub, Receiver};
+use sp_core::storage::{StorageData, StorageKey};
 use sp_runtime::traits::Block as BlockT;
 
+mod registry;
+
+use registry::Registry;
+
+#[cfg(test)]
+mod tests;
+
+/// A type of a message delivered to the subscribers
+pub type StorageNotification<Hash> = (Hash, StorageChangeSet);
```

Comment:
```suggestion
pub struct StorageNotification<Hash> { hash: Hash, changes: StorageChangeSet }
```

Would make it nicer to work with the type.